src/open_inwoner/cms/cases/views/services.py

```diff
@@ -186,31 +186,41 @@ def get_cases(self) -> list[ZaakWithApiGroup]:
 
         return cases_with_api_group
 
-    def _get_cases_for_api_group(self, group: ZGWApiGroupConfig):
+    def _get_cases_for_api_group(self, group: ZGWApiGroupConfig) -> list[Zaak]:
         raw_cases = group.zaken_client.fetch_cases(
             **get_user_fetch_parameters(self.request)
         )
-        preprocessed_cases = self.resolve_cases(raw_cases, group)
-        return preprocessed_cases
+        resolved_cases = self.resolve_cases(raw_cases, group)
+
+        filtered_cases = [
+            case for case in resolved_cases if case.status and is_zaak_visible(case)
+        ]
+        filtered_cases.sort(key=lambda case: case.startdatum, reverse=True)
+        return filtered_cases
 
     def resolve_cases(
         self,
         cases: list[Zaak],
         group: ZGWApiGroupConfig,
     ) -> list[Zaak]:
+        resolved_cases = []
         with parallel(max_workers=self._thread_limits["resolve_case_list"]) as executor:
             futures = [
                 executor.submit(self.resolve_case, case, group) for case in cases
             ]
```

Comment:
It could be useful to log the identity of the case that failed to resolve. To this end, we could use a `dict` comprehension instead of a `list` as `futures`:

```
futures = {
    executor.submit(self.resolve_case, case, group): {"case": case, "api_group": group} for case in cases
}
```

Then the log could be:

```
except BaseException:
    case = task["case"]
    group = task["api_group"]
    logger.exception("Error while resolving case {case} with API group {group}".format(case=case, group=group))
```